### ISSUE TITLE
Support for validating sub-schemas?

### ISSUE THREAD

**Issue by AliciaSchep** [ropensci/jsonvalidate#18]
I was wondering if this package might also support validating sub-schemas, e.g. giving a reference within a schema to validate, as ajv  (which looks like it will be supported soon, thanks @karawoo , @ijlyttle, @richfitz ! ) supports that kind of validation. It can be useful to verify that a specific part of a json is valid, for example when building up a json bit by bit. (I'm not actually sure `is-my-json-valid` doesn't support it, but not obvious to me how to go about it using that package).

Here is a little example using ajv within R (using the `use-ajv` branch of `jsonvalidate`) to validate just the 'Axis' definition within the Vega-Lite schema:

```r
library(jsonvalidate)

vl_schema <- readr::read_file(
  "https://raw.githubusercontent.com/vegawidget/vegawidget/8d95a9ac74ea0e590259a88b9d5fa8e127f54fb9/inst/schema/vega-lite/v3.3.0.json")

jsonvalidate:::env$ct$eval(
  sprintf("subTest = new Ajv({verbose: true}).addMetaSchema(AjvSchema6).addSchema(%s,'vega-lite');",
          vl_schema)
)

ref <- "#/definitions/Axis"

jsonvalidate:::env$ct$eval(sprintf("subVal = subTest.getSchema('%s');", ref))

test_json_positive = jsonlite::toJSON(
  list(domain = TRUE), 
  auto_unbox = TRUE, 
  null = "null")

res_yes <- jsonvalidate:::env$ct$call("subVal", V8::JS(test_json_positive))
res_yes #Should be TRUE

test_json_negative = jsonlite::toJSON(
  list(domain = 1), 
  auto_unbox = TRUE, 
  null = "null")

res_no <- jsonvalidate:::env$ct$call("subVal", V8::JS(test_json_negative))
res_no #should be FALSE
```

In terms of how this would fit into the current `jsonvalidate` API, I would propose that `json_validator` and `json_validate` can both get an optional `reference` parameter.  If specified then a validator/validation for that reference is made rather than the full json schema.  

If that addition (with either the API I am proposing or an alternate approach for incorporating) would be welcome in this package, I'd be happy to put together a PR! Just want to check first whether such an addition might fit into this package or not...

**Comment by richfitz** [comment]
Yes, I would be interested in a PR for this if you're happy to put one together.  If you could add a section to the vignette too showing how this is useful that would be fab

**Comment by AliciaSchep** [comment]
Awesome, thanks, will start thinking a bit more on the implementation and hopefully have a PR sometime in the next couple of weeks (with an example for vignette!).

**Comment by richfitz** [comment]
Fixed in #19
